src/main/java/tdl/client/queue/QueueBasedImplementationRunner.java

```diff
@@ -147,46 +111,16 @@ private static ProcessingRules createDeployProcessingRules() {
         }
     }
 
-    static class MessageProcessingThread extends Thread {
-        private final AtomicBoolean shouldContinue;
-        private final RemoteBroker remoteBroker;
-        private List<Request> requests;
-        private Audit audit;
-
-        MessageProcessingThread(RemoteBroker remoteBroker, Audit audit, List<Request> requests) {
-            this.remoteBroker = remoteBroker;
-            this.audit = audit;
-            this.requests = requests;
-
-            shouldContinue = new AtomicBoolean(true);
-        }
-
-        @Override
-        public void run() {
-            while (shouldContinue.get()) {
-                try {
-                    List<Request> requestsBatch = remoteBroker.receive();
-                    requests.addAll(requestsBatch);
-                } catch (Exception e) {
-                    logToConsole("        QueueBasedImplementationRunner run [error]");
-                    String message = "There was a problem processing messages";
-                    LOGGER.error(message, e);
-                    audit.logException(message, e);
-                }
-            }
-        }
-
-        void signalStop() {
-            shouldContinue.set(false);
-        }
-    }
-
     public void run() {
         logToConsole("        QueueBasedImplementationRunner run [start]");
         audit.logLine("Starting client");
         requests.clear();
         try {
-            messageProcessingThread.join(750);
+            requests.clear();
+            remoteBroker = new RemoteBroker(config.getHostname(), config.getPort(), config.getUniqueId(), config.getRequestTimeoutMillis());
```

Comment:
I think I see where the problem is.
As it stands, the current RemoteBroker abstraction does not work for the SQS implementation.
My suggestion is to completely deconstruct this abstraction and all other abstractions that do not work well and use the AWS client directly inside this class.

I am curious to see how it looks once we inline everything here.
Then we can re-arrange things so that it works for SQS.